### ISSUE TITLE
`left in right` documentation is incorrect?

### ISSUE THREAD

**Issue by tjchambers** [elixir-ecto/ecto#4176]
### Elixir version

1.14.4

### Database and Version

MySQL 8.0.30

### Ecto Versions

3.10.1

### Database Adapter and Versions (postgrex, myxql, etc)

myxql 0.6.3

### Current behavior

[Current Docs](https://hexdocs.pm/ecto/Ecto.Query.API.html#in/2)

```
from c in Comment, where: c.post_id in subquery(
  from(p in Post, where: p.created_at > ^since)
)
```

...is the current documented example. I was curious to see how this "inferred" that the `in` was deducing that the `id` column of Post was the comparison. 

Emulating that in my DB with similar schema in my db I tried:

```
from c in Sct.Internal.Tenant, where: c.org_id in subquery(
   from(x in Sct.Network.Collective)
)
```
 which returns the query
 
```
 #Ecto.Query<from t0 in Sct.Internal.Tenant,
 where: t0.org_id in subquery(#Ecto.Query<from c0 in Sct.Network.Collective>)>
```

However fetching the data results in ...

```
from c in Sct.Internal.Tenant, where: c.org_id in subquery(
   from(x in Sct.Network.Collective)
) |> Sct.Repo.all
```
** (Ecto.Query.CompileError) `Sct.Repo.all(c.org_id in subquery(from(x in Sct.Network.Collective)))` is not a valid query expression.

From that I deduced that the example was missing proper parentheses (even though it did interpret the Query without them. So I added parentheses>>>

```
from(c in Sct.Internal.Tenant, where: c.org_id in subquery(from(x in Sct.Network.Collective))) |> Sct.Repo.all
```

... which led me to the answer I was investigating

```
** (Ecto.QueryError) subquery must return a single field in order to be used on the right-side of `in` in query:

from t0 in Sct.Internal.Tenant,
  where: t0.org_id in subquery(
  #Ecto.Query<from c0 in Sct.Network.Collective, select: %{
  id: c0.id,
  name: c0.name,
  active: c0.active,
  source: c0.source,
  community_id: c0.community_id,
  eusage: c0.eusage,
  expose: c0.expose,
  keyword: c0.keyword,
  notes: c0.notes,
  permalink: c0.permalink,
  type: c0.type,
  tenant_id: c0.tenant_id,
  created_at: c0.created_at,
  updated_at: c0.updated_at
}>
),
  select: t0

    (elixir 1.14.4) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir 1.14.4) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto 3.10.1) lib/ecto/repo/queryable.ex:211: Ecto.Repo.Queryable.execute/4
    (ecto 3.10.1) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
```

### Expected behavior

So to summarize:

The query as written is most likely missing 2 elements:

- The initial `from` should have an outer parentheses around the whole example.
- The subquery MUST return a column for comparison using `select` (which is significant to this example)

Corrected query (IMHO):

```
from(c in Comment, where: c.post_id in subquery(
  from(p in Post, where: p.created_at > ^since, select: p.id)
)
)
```

**Comment by v0idpwn** [comment]
Hello, @tjchambers, thanks for the report!

The example is indeed wrong about not selecting a field. I opened a PR fixing it: https://github.com/elixir-ecto/ecto/pull/4177

Regarding the parenthesis, the issue that you have faced is related to the pipeline operator associating to part of the query expression. You could, for example, assign it to a variable and use `Repo.all/1` on it. So I'm not sure we should **fix** it. I think we have similar code in many other examples.
